pkg/controller/util.go

```diff
@@ -619,7 +619,8 @@ func ValidateCanCloneSourceAndTargetSpec(sourceSpec, targetSpec *v1.PersistentVo
 		targetVolumeMode = v1.PersistentVolumeBlock
 	}
 	if sourceVolumeMode != targetVolumeMode {
-		return errors.New("source and target volume modes do not match")
+		return fmt.Errorf("source volumeMode (%v) and target volumeMode (%v) do not match",
```

Comment:
So I would use %s instead of %v here. %s prints a string, while %v prints an entire object. In this case the type is actually a string, so %v and %s will return the same result. But if you have a more complex struct or something, then %v will print everything in the struct. Depending on what you are showing this may or may not be what you want.

https://golang.org/pkg/fmt/ will explain it better than I can.